src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/ApiController.cs

```diff
@@ -63,19 +70,43 @@ public class ApiController : Controller
                 return this.ChallengeOrForbid();
             }
 
+            var result = await LuceneQueryApi(indexName, query, parameters);
+
+            return new ObjectResult(result);
+        }
+
+        [HttpPost]
+        [Route("documents")]
+        public async Task<IActionResult> DocumentsPost(
+            [FromForm] string indexName,
+            [FromForm] string query,
+            [FromForm] string parameters)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.QueryLuceneApi))
+            {
+                return this.ChallengeOrForbid();
+            }
+
+            var result = await LuceneQueryApi(indexName, query, parameters);
+
+            return new ObjectResult(result);
+        }
+
+        private async Task<Queries.IQueryResults> LuceneQueryApi(string indexName, string query, string parameters, bool returnContentItems = false)
+        {
             var luceneQuery = new LuceneQuery
             {
                 Index = indexName,
-                Template = query
+                Template = query,
+                ReturnContentItems = returnContentItems
             };
 
             var queryParameters = parameters != null ?
                 JsonConvert.DeserializeObject<Dictionary<string, object>>(parameters)
                 : new Dictionary<string, object>();
 
             var result = await _luceneQuerySource.ExecuteQueryAsync(luceneQuery, queryParameters);
```

Comment:
Because there is no ther await before, here you can just return

    return _luceneQuerySource.ExecuteQueryAsync(luceneQuery, queryParameters);

And then remove the async modifier above line 95

Reply:
Yes ,  Thank you